observatory-platform/observatory/platform/utils/workflow_utils.py

```diff
@@ -1320,3 +1308,78 @@ def make_table_name(
         new_table_name = f"{table_id}{table_date.strftime('%Y%m%d')}"
 
     return new_table_name
+
+
+def get_chunks(*, input_list: List[Any], chunk_size: int = 8) -> List[Any]:
+    """Generator that splits a list into chunks of a fixed size.
+
+    :param input_list: Input list.
+    :param chunk_size: Size of chunks.
+    :return: The next chunk from the input list.
+    """
+
+    n = len(input_list)
+    for i in range(0, n, chunk_size):
+        yield input_list[i : i + chunk_size]
+
+
+class AsyncHttpFileDownloader:
```

Comment:
At the moment we are using third party code to download files, specifically the function `get_file` in `data_utils.py`.
* https://github.com/The-Academic-Observatory/observatory-platform/blob/develop/observatory-platform/observatory/platform/utils/data_utils.py
* https://github.com/The-Academic-Observatory/observatory-platform/blob/develop/observatory-platform/observatory/platform/utils/progbar_utils.py
* https://github.com/The-Academic-Observatory/observatory-platform/blob/87a78df26a44f0d890a87fc2ba50e3ba1fd653fc/observatory-platform/observatory/platform/utils/file_utils.py#L91-L111: this function should be removed as well, it isn't from our project and should be in data_utils.py if it is going to stay.
* https://github.com/The-Academic-Observatory/observatory-platform/blob/87a78df26a44f0d890a87fc2ba50e3ba1fd653fc/observatory-platform/observatory/platform/utils/file_utils.py#L59-L88: this function should be removed as well, it isn't from our project and should be in data_utils.py if it is going to stay.

We could almost completely replace it with your code, but there is one thing missing, which is the ability to verify if a downloaded file has a particular file hash (you'll see the parameters `file_hash` and `hash_algorithm` it in `get_file`), which is important to make sure that a partly downloaded file is not accepted as a fully downloaded file after a task crashes and restarts. 

Would you be able to add `file_hash=None` as an optional parameter?
* We could just use md5 for now.
* If `file_hash` hash is supplied, then validate the file.
  * If the validation fails, then re-download the file.
  * If the validation succeeds, skip downloading the file. 
* If no `file_hash` is supplied:
  * Is the default behaviour to overwrite existing files?

We would also then need to replace instances of `get_file` throughout the code.

This would also mean that we could remove the following dependencies from requirements.txt:
```
# To remove in future
numpy>=1.20.3,<2
six>=1.16.0,<2
```